src/internal/modules/charge-information/pre-handlers.js

```diff
@@ -126,14 +126,31 @@ const loadChargeVersion = async request => {
   }
 };
 
+const flattenAdditionalChargesProperties = ({ additionalCharges, ...element }) => {
+  if (additionalCharges) {
+    const { supportedSource, isSupplyPublicWater } = additionalCharges;
+    element.isAdditionalCharges = true;
+    element.isSupportedSource = !!supportedSource;
+    element.isSupplyPublicWater = isSupplyPublicWater;
+    const { id, name } = supportedSource;
+    if (id) {
+      element.supportedSourceId = id;
+    }
+    if (name) {
```

Comment:
As above. Not sure if the if statement is needed. If it is, leave it

Reply:
I didn't want the property to exist in the element even if it was set to undefined as for the id.